src/authorization/WebAclReader.ts

```diff
@@ -20,16 +18,19 @@ import type { PermissionReaderInput } from './PermissionReader';
 import { PermissionReader } from './PermissionReader';
 import type { AclPermission } from './permissions/AclPermission';
 import { AclMode } from './permissions/AclPermission';
-import { AccessMode } from './permissions/Permissions';
 import type { PermissionSet } from './permissions/Permissions';
+import { AccessMode } from './permissions/Permissions';
 
-const modesMap: Record<string, keyof AclPermission> = {
-  [ACL.Read]: AccessMode.read,
-  [ACL.Write]: AccessMode.write,
-  [ACL.Append]: AccessMode.append,
-  [ACL.Control]: AclMode.control,
+// Maps ACL modes to their associated general modes.
+const modesMap: Record<string, Readonly<(keyof AclPermission)[]>> = {
+  [ACL.Read]: [ AccessMode.read ],
+  [ACL.Write]: [ AccessMode.append, AccessMode.write ],
+  [ACL.Append]: [ AccessMode.append ],
+  [ACL.Control]: [ AclMode.control ],
 } as const;
 
+type AclSet = { target: Store; parent?: Store };
```

Comment:
Is `parent` always a container, and should we name it as such? (Yeah I'm probably confused.)

Reply:
This is for `POST`, isn't it?

Reply:
`parent` is always going to be the parent container yes. It's used for all cases where we need the parent permissions, which are all create cases (so POST/PATCH/PUT) and delete cases.

In the future this might get extended to a more generic interface if readers need to find permissions for multiple resources.